Tests/Common/JoinAcceptResponse.cs

```diff
@@ -8,16 +8,19 @@ namespace LoRaWan.Tests.Common
     internal sealed record JoinAcceptResponse
```

Comment:
I would consider moving this as a private record of `SimulatedDevice` where it's only used in one place. It avoids having a lot of stand-alone types floating at the root that are more private implementation details (for now) and which don't need the same design scrutiny applied. For example, you could define it simply as:

```c#
private sealed record JoinAcceptResponse(
    [property: JsonPropertyName("pdu")] string Pdu,
    [property: JsonPropertyName("DevEui")] string DevEuiString);
```